samcli/lib/sync/infra_sync_executor.py

```diff
@@ -272,6 +274,16 @@ def _auto_skip_infra_sync(
             LOG.debug("The current template is different from the last deployed version, we will not skip infra sync")
             return False
 
+        try:
+            current_stack_params = self._get_stack_parameters(stack_name)
+        except ClientError as ex:
+            LOG.debug("Unable to fetch stack Parameters from stack with name %s", stack_name, exc_info=ex)
+            return False
```

Comment:
I would move this into new function as well since this is purely related to that functionality